feat(cli): add optional difficulty tag to problems

tony-buildd opened the pull request:
- Allow specifying difficulty (easy/medium/hard) as an optional positional argument in [add] command
- Support adding difficulty in `nextup` command
- Display color-coded difficulty tags in `list`, `inprogress`, and `nextup` outputs
- Add "Difficulty" column to `mastered` table
- Update help examples to show usage

tony-buildd committed:
feat(cli): add optional difficulty tag to problems
- Allow specifying difficulty (easy/medium/hard) as an optional positional argument in [add] command
- Support adding difficulty in `nextup` command
- Display color-coded difficulty tags in `list`, `inprogress`, and `nextup` outputs
- Add "Difficulty" column to `mastered` table
- Update help examples to show usage

HayesBarber commented on srl/commands/inprogress.py:
```diff
@@ -15,9 +15,21 @@ def add_subparser(subparsers):
+        problem_lines = []
+        for name, difficulty in in_progress:
+            diff_tag = ""
+            if difficulty:
```
Difficulty color shows up a couple times. I would make it a function.

HayesBarber reviewed (changes requested):
I like the concept here, just a few small comments. Also, quite a few tests are failing. I can work on a GHA to run tests automatically on PRs.

tony-buildd committed:
refactor: extract difficulty color logic to utils

- Extract `get_difficulty_tag` helper in `srl/utils.py`
- Update `srl/commands/inprogress.py` to use helper
- Update `srl/commands/list_.py` to use helper
- Ensure consistent difficulty tag formatting across commands

TonyNguyenVn17 commented on srl/commands/list_.py:
```diff
@@ -66,15 +79,30 @@ def get_due_problems(limit=None) -> list[str]:
+    for name, info in list(next_up.items())[: limit or 3]:
+        # Next Up items in file don't store difficulty usually, but let's check if we save it there
+        # The current nextup implementation saves {"added": date}, but we might add difficulty there too.
+        # For now, let's assume difficulty might be there or None.
```
Oh i got it. I forgot I did it because I was developed another feat in another branch and doing it help me prevent conflict. I also realize the line below is self explanatory so i don't need comment too

tony-buildd committed:
refactor: cleanup comments and use difficulty helper in nextup

- Remove outdated comment in `srl/commands/list_.py`
- Refactor `srl/commands/nextup.py` to use `get_difficulty_tag`
- Ensure consistent difficulty tag rendering

tony-buildd committed:
refactor: remove unnecessary comments project-wide

- Remove redundant comments in `srl/commands/list_.py`
- Remove redundant comments in `srl/commands/add.py`
- Remove redundant comments in `srl/commands/audit.py`
- Improve code cleanliness

HayesBarber commented:
Closing for now. May revisit this in the future.
